### ISSUE TITLE
FULL OUTER JOIN leads to segfault

### ISSUE THREAD

**Issue by osbertngok** [yandex/ClickHouse#844]
Env:

Mac, docker image from latest https://hub.docker.com/r/yandex/clickhouse-server/

query:
```
SELECT bla, blo, campaignID FROM
(SELECT 1 AS bla, campaignID FROM system.one ARRAY JOIN ['c1', 'c2', 'c3'] AS campaignID) AS TB1
ANY FULL OUTER JOIN
(SELECT 1 AS blo, campaignID FROM system.one ARRAY JOIN ['c4', 'c5', 'c6'] AS campaignID) AS TB2
USING campaignID
```

Error:
```
ch1_1   | 2017.06.01 07:53:30.011516 [ 4 ] <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: POST, Address: 172.19.0.1:48306, User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/58.0.3029.96 Safari/537.36
ch1_1   | 2017.06.01 07:53:30.011618 [ 4 ] <Trace> HTTPHandler: Request URI: /?add_http_cors_header=1&log_queries=1&database=system&max_result_rows=500&result_overflow_mode=throw
ch1_1   | 2017.06.01 07:53:30.012386 [ 4 ] <Debug> executeQuery: (from 172.19.0.1:48306)  /*TABIX_QUERY_ID_BvLMXVqM*/ SELECT bla, blo, campaignID FROM (SELECT 1 AS bla, campaignID FROM system.one ARRAY JOIN ['c1', 'c2', 'c3'] AS campaignID) AS TB1 ANY FULL OUTER JOIN (SELECT 1 AS blo, campaignID FROM system.one ARRAY JOIN ['c4', 'c5', 'c6'] AS campaignID) AS TB2 USING campaignID   
ch1_1   | 2017.06.01 07:53:30.014932 [ 4 ] <Trace> InterpreterSelectQuery: FetchColumns -> Complete
ch1_1   | 2017.06.01 07:53:30.015079 [ 4 ] <Trace> InterpreterSelectQuery: FetchColumns -> Complete
ch1_1   | 2017.06.01 07:53:30.016867 [ 4 ] <Debug> executeQuery: Query pipeline:
ch1_1   | CreatingSets
ch1_1   |  Lazy
ch1_1   |  Union
ch1_1   |   Expression
ch1_1   |    Expression
ch1_1   |     Expression
ch1_1   |      Expression
ch1_1   |       Expression
ch1_1   |        One
ch1_1   |   Expression
ch1_1   |    Expression
ch1_1   |     NonJoined
ch1_1   | 
ch1_1   | 2017.06.01 07:53:30.018112 [ 4 ] <Trace> CreatingSetsBlockInputStream: Creating join. 
ch1_1   | 2017.06.01 07:53:30.018933 [ 4 ] <Trace> InterpreterSelectQuery: FetchColumns -> Complete
ch1_1   | 2017.06.01 07:53:30.020552 [ 4 ] <Debug> CreatingSetsBlockInputStream: Created. Join with 3 entries from 3 rows. Read 1 rows, 0.000 MiB in 0.002 sec., 521 rows/sec., 0.000 MiB/sec.
ch1_1   | 2017.06.01 07:53:30.021323 [ 8 ] <Error> BaseDaemon: ########################################
ch1_1   | 2017.06.01 07:53:30.021429 [ 8 ] <Error> BaseDaemon: (from thread 7) Received signal Segmentation fault (11).
ch1_1   | 2017.06.01 07:53:30.021462 [ 8 ] <Error> BaseDaemon: Address: NULL pointer.
ch1_1   | 2017.06.01 07:53:30.032100 [ 8 ] <Error> BaseDaemon: 1. /usr/bin/clickhouse-server(DB::Block::bytes() const+0x27) [0x17fb2a7]
ch1_1   | 2017.06.01 07:53:30.032575 [ 8 ] <Error> BaseDaemon: 2. /usr/bin/clickhouse-server(Poco::ThreadImpl::runnableEntry(void*)+0xa5) [0x33fb245]
ch1_1   | 2017.06.01 07:53:30.032649 [ 8 ] <Error> BaseDaemon: 3. /lib/x86_64-linux-gnu/libpthread.so.0(+0x76ba) [0x7fe0743586ba]
ch1_1   | 2017.06.01 07:53:30.032692 [ 8 ] <Error> BaseDaemon: 4. /lib/x86_64-linux-gnu/libc.so.6(clone+0x6d) [0x7fe07397982d]
ch1_1   | 2017.06.01 07:53:36.400212 [ 6 ] <Trace> SystemLog (system.query_log): Flushing query log
ch1_1   | 2017.06.01 07:53:36.418921 [ 6 ] <Trace> system.query_log (Data): Renaming tmp_20170601_20170601_3_3_0.
```

Let me know if you need further information. Thanks.

**Comment by ztlpn** [comment]
Strange, because this bug was fixed in v1.1.54236 which is the current version at hub.docker.com. Could you check the server version and re-execute `docker pull yandex/clickhouse-server` to be sure?

**Comment by osbertngok** [comment]
My apologies, looks like I haven't pulled the image for a while.
I was using sha256:37ff52724dbad4b8dfdbe74b06e2cbc2b8ce5997621700bbb4b15bbf1febfc9b, and the current latest sha256:4b035102d5c996aec31de346b24768adf99a83af7dc4e051e41c96c432e8ec32 works. Please feel free to close this issue.
